src/pal/src/file/find.cpp

```diff
@@ -777,18 +778,29 @@ static int FILEGlobFromSplitPath( CPalThread *pthrCurrent,
                                   glob_t *pgGlob )
 {
     int  Ret;
-    char Pattern[MAX_LONGPATH];
-    char EscapedPattern[2*MAX_LONGPATH];
+    PathCharString PatternPS;
+    PathCharString EscapedPatternPS;
+    char * Pattern;
+    int length = 0;
+    char * EscapedPattern;
 
     TRACE("We shall attempt to glob from components [%s][%s][%s]\n",
           dir?dir:"NULL", fname?fname:"NULL", ext?ext:"NULL");
 
-    FILEMakePathA( Pattern, MAX_LONGPATH, dir, fname, ext );
+    if (dir) length = strlen(dir);
+    if (fname) length += strlen(fname);
+    if (ext) length += strlen(ext);
+    
+    Pattern = PatternPS.OpenStringBuffer(length);
+    FILEMakePathA( Pattern, length+1, dir, fname, ext );
+    PatternPS.CloseBuffer(length);
     TRACE("Assembled Pattern = [%s]\n", Pattern);
 
     /* special handling is needed to handle the case where
         filename contains '[' and ']' */
+    EscapedPattern = EscapedPatternPS.OpenStringBuffer(length);
```

Comment:
The length here is not enough. The FILEEscapeSquareBrackets needs one more character space available for every occurence of `[` and `]` characters in the pattern. We can either make the buffer twice the size of the pattern or modify the FILEEscapeSquareBrackets to get the stack string pointer as a parameter do two passes over the string or something like that.